Sources/App/Controllers/Karma/KarmaController+Slack.swift

```diff
@@ -43,59 +44,55 @@ extension KarmaController {
-        }
+            }
     }
 }
 
 extension KarmaController: SlackResponder {
 
-    func handle(message: Message) throws {
-        guard let sendingUser = message.sender else { return }
+    func handle(incomingMessage: SlackKitIncomingMessage) throws {
+
+        let karmaChanges = karmaParser.karmaAdjustments(from: incomingMessage.text)
 
-        let karmaMessages = karmaParser.karmaMessages(from: message.text)
+        let slack = self.slack
+        let statusRepository = self.karmaStatusRepository
+        let historyRepository = self.karmaHistoryRepository
+        let log = self.log
 
-        try karmaMessages.forEach { karmaMessage in
-            let slack = self.slack
-            let statusRepository = self.karmaStatusRepository
-            let historyRepository = self.karmaHistoryRepository
-            let log = self.log
+        try karmaChanges.forEach { change in
 
-            guard karmaMessage.user != sendingUser else {
+            guard change.user != incomingMessage.sender else {
                 let errorMessage = "You can't adjust karma for yourself! "
-                try slack.send(message: message.response(with: errorMessage), onlyVisibleTo: sendingUser)
+                try slack.send(message: SlackKitResponse(to: incomingMessage, text: errorMessage), onlyVisibleTo: incomingMessage.sender)
                 return
             }
 
             // Save history record
-            let karmaHistory = KarmaSlackHistory(karmaCount: karmaMessage.count, fromUser: sendingUser, karmaReceiver: karmaMessage.user, channel: message.channelID.id)
-            historyRepository.save(history: karmaHistory)
+            let karmaHistory = KarmaSlackHistory(karmaCount: change.count, fromUser: incomingMessage.sender, karmaReceiver: change.user, channel: incomingMessage.channelID.id)
+            historyRepository
+                .save(history: karmaHistory)
                 .catch {
                     log.error("Could not save history \($0)")
                 }
 
             // Update karma
-            statusRepository.find(id: karmaMessage.user)
-                .unwrap(or: Abort(.notFound))
-                .flatMap { storedKarma in
-                    storedKarma.count += karmaMessage.count
-                    return statusRepository.save(karma: storedKarma)
-                }.catchFlatMap { _ in
-                    statusRepository.save(karma: karmaMessage.statusData())
-                }.thenThrowing { karma -> Void in
-                    let response = message.response(with: karmaMessage.slackUser(),
-                                                    attachments: [karmaMessage.slackAttachment(with: karma.count)])
-                    try slack.send(message: response)
+            statusRepository
+                .find(id: change.user)
+                .flatMap {
+                    let karmaStatus = $0 ?? KarmaStatus(id: change.user, count: 0)
```

Comment:
Oh I see what you are doing here. Nice find to clean up one block

Reply:
It's a little code golf-y, I'll admit! I just got inspired to go further though...

```
statusRepository.save(karma: KarmaStatus(id: change.user, count: ($0?.count ?? 0) + change.count))
```

I actually think this is more readable!